connect/runtime/src/test/java/org/apache/kafka/connect/util/KafkaBasedLogTest.java

```diff
@@ -234,43 +242,42 @@ public void testReloadOnStartWithNoNewRecordsPresent() throws Exception {
         });
 
         store.start();
+        expectStart();
 
         assertEquals(CONSUMER_ASSIGNMENT, consumer.assignment());
         assertEquals(7L, consumer.position(TP0));
         assertEquals(7L, consumer.position(TP1));
 
         store.stop();
+        expectStop();
 
-        assertFalse(Whitebox.<Thread>getInternalState(store, "thread").isAlive());
+        assertFalse(store.thread.isAlive());
         assertTrue(consumer.closed());
-        PowerMock.verifyAll();
     }
 
     @Test
     public void testSendAndReadToEnd() throws Exception {
-        expectStart();
         TestFuture<RecordMetadata> tp0Future = new TestFuture<>();
         ProducerRecord<String, String> tp0Record = new ProducerRecord<>(TOPIC, TP0_KEY, TP0_VALUE);
-        Capture<org.apache.kafka.clients.producer.Callback> callback0 = EasyMock.newCapture();
-        EasyMock.expect(producer.send(EasyMock.eq(tp0Record), EasyMock.capture(callback0))).andReturn(tp0Future);
+        ArgumentCaptor<org.apache.kafka.clients.producer.Callback> callback0 =
+            ArgumentCaptor.forClass(org.apache.kafka.clients.producer.Callback.class);
+        when(producer.send(eq(tp0Record), callback0.capture())).thenReturn(tp0Future);
         TestFuture<RecordMetadata> tp1Future = new TestFuture<>();
         ProducerRecord<String, String> tp1Record = new ProducerRecord<>(TOPIC, TP1_KEY, TP1_VALUE);
-        Capture<org.apache.kafka.clients.producer.Callback> callback1 = EasyMock.newCapture();
-        EasyMock.expect(producer.send(EasyMock.eq(tp1Record), EasyMock.capture(callback1))).andReturn(tp1Future);
+        ArgumentCaptor<org.apache.kafka.clients.producer.Callback> callback1 =
+            ArgumentCaptor.forClass(org.apache.kafka.clients.producer.Callback.class);
+        when(producer.send(eq(tp1Record), callback1.capture())).thenReturn(tp1Future);
 
         // Producer flushes when read to log end is called
-        producer.flush();
-        PowerMock.expectLastCall();
-
-        expectStop();
-
-        PowerMock.replayAll();
+        doNothing().when(producer).flush();
```

Comment:
This isn't necessary anymore; we don't have to establish no-op expectations for void methods.

It does technically provide some guarantees that `producer::flush` has been invoked by the end of the test since we'd see an `UnnecessaryStubbingException` otherwise, but it'd be better to be more explicit about that by adding `verify(producer).flush()` sometime after the call to `store::readToEnd`. We probably don't want to put it _directly_ after that call since it's made inside the mock consumer, but it should be fine to add it after the line with `readEndFutureCallback::get`.